Responsive CSS for Outfitting screen

MurtaghJustin opened the pull request:
I've responsived up the outfitting screen and fixed a small issue on really old iPhones on the ship selection screen.

Everything on the outfitting screen should look ok now on desktops down on to the original iPhone.

One thing I didn't change was the summary table, I started, but ended up putting those changes in a separate branch so I can work on them some more. For now, that table becomes scrollable horizontally on smaller devices (once it can't shrink down anymore). It's not ideal, but it's better than before at least :).

MurtaghJustin committed:
Merge pull request #1 from cmmcleod/master

If I'm doing this right, this will take updates to my fork

MurtaghJustin committed:
More responsive outfitting screen (still work to do though).

MurtaghJustin committed:
Other responsive changes. Outfitting screen now works pretty well on phone-sized displays. Summary is currently just a horizontal-scrolling thing (because it's a table atm) but everything else scales down and collapses pretty nicely.

MurtaghJustin committed:
Fixes for sub-350px screens (eg: really old iPhones)

cmmcleod commented:
Cool stuff!

Can you please move the media tags to responsive.less and refactor to be similar to `.tablet()`?

Also remove the `screen` clause such that if a browser desktop window is resized it will adjust in a similar manner.

Also, I think this are some whitespace /spaces diff issues (4 spaces vs 2).

MurtaghJustin commented:
Yeah, my IDE is configured to use 4 spaces and I keep forgetting to change it :). I might just switch for this project. I'll take a look at the other stuff tonight after work. Although, re-sizing the browser window should already work just fine (works for me in Chrome and IE), but I'll double-check it :).

cmmcleod commented:
Just ignore the `screen` comment.. I was thinking of `width` vs `device-width`, etc

MurtaghJustin committed:
Refactored responsive stuff.

MurtaghJustin committed:
4-to-2 spaces :)

MurtaghJustin commented:
I've refactored the responsive stuff to be like the tablet stuff. Now can just use .tablet({ property: value; }); or .smallPhone({ property: value; }); wherever.

I've removed the minimums from these responsive things because in almost all cases you'll want the things you change for tablets to also apply to large phones, things for medium phones should also be for small phones, etc.

Still can have unique things for specific sizes just by ordering the calls to .mediumPhone(), .smallPhone(), .largePhone(), etc (lowest wins).

We can change it so styles don't flow through - but it will mean writing a lot more css, eg:
.tablet({
prop1: 0;
prop2: 1;
});
.smallPhone({
prop3: 1;
});

_Would become_:
.tablet({
prop1: 0;
prop2: 1;
});
.largePhone({
prop1: 0;
prop2: 1;
})
.medPhone({
prop1: 0;
prop2: 1;
})
.smallPhone({
prop1: 0;
prop2: 1;
prop3: 1;
});

I also changed my formatting to be 2 spaces instead of 4 (and auto-format fixed up a couple of other small formatting things).

MurtaghJustin committed:
Font-size bump for tablets and phones on the outfitting screen. Seems to make life a bit easier. Also added a small-tablet size option which will fix a tiny visual bug on the outfitting screen with long ship names. Fixed a few other minor visual bugs. Also made clickable menu items bigger on tablets and phones.

MurtaghJustin commented:
Added another commit, description for that pretty much tells the story. I did a bunch of testing and changed/fixed the things that annoyed my on my tablet and phone :). I think I have now tried each and every ship at all the different screen sizes.

cmmcleod commented:
:+1:
